### PR TITLE
Etcd backup app

### PR DESCRIPTION
**Pull request by mszostok**
Confirm these statements before you submit your pull request:

- [x] I have read and submitted the required [Contributor Licence Agreements](https://github.com/kyma-project/community/blob/master/CONTRIBUTING.md#agreements-and-licenses).
- [x] This pull request follows the contributing guidelines, recommended Git workflow, and templates.
- [x] I have tested my changes.
- [x] I have updated the relevant documentation.
---

**Description**

Changes proposed in this pull request:

- Add the Go application which will be used in [CronJob](https://kubernetes.io/docs/concepts/workloads/controllers/cron-jobs/) to make periodic etcd backups.
- App has the following capabilities/responsibilities:
  - create an EtcdBackup CR (trigger backup process )
  - watch status on created EtcdBackup CR
  - clean-up created EtcdBackup CR
  - save in k8s ConfigMap path to the  ABS backup file from the last success
  - exit with correct status code

This is more robust solution than proposed by [coreos](https://github.com/coreos/etcd-operator/tree/master/doc/user/backup_cronjob)
The cleaner service (rotate old backup files) will be added in separate PR. The preview you can see here: https://github.com/mszostok/kyma/compare/etcd-backup-app...mszostok:etcd-backup-cleaner-app?expand=1. I didn't make a one PR because it will be about 3k LOC, so it will be easier for the reviewer to review it separately.

### CODE REVIEW

File: tools/etcd-backup/README.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
+# Etcd Backup
+
+## Overview
+
+The etcd-backup trigger a backup process of etcd-cluster via etcd-backup-operator.
```
Comment (dpolitesap) [inline tools/etcd-backup/README.md]: The etcd-backup triggers a backup process of etcd-cluster using the etcd-backup-operator.

File: tools/etcd-backup/README.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+For the description of the available environment variables, see the **Use environment variables** section.
+
+### Use environment variables
+Use the following environment variables to configure the application:
+
+| Name | Required | Default | Description |
+|-----|---------|--------|------------|
+| **APP_LOGGER_LEVEL** | No | `info` | Show detailed logs in the application. |
+| **APP_KUBECONFIG_PATH** | No |  | The path to the `kubeconfig` file that you need to run an application outside of the cluster. |
+| **APP_WORKING_NAMESPACE** | Yes |  | The name of the Namespace where Etcd Backup application is executed. |
```
Comment (dpolitesap) [inline tools/etcd-backup/README.md]: The Namespace where the Etcd Backup application is executed.

File: tools/etcd-backup/README.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+| **APP_ABS_CONTAINER_NAME** | Yes |  | The Azure Blob Storage container name where the backup should be saved. |
```
Comment (dpolitesap) [inline tools/etcd-backup/README.md]: The Azure Blob Storage container to store the backup.

File: tools/etcd-backup/README.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+| **APP_ABS_SECRET_NAME** | Yes |  | The name of the secret object that stores the Azure storage credential. |
+| **APP_BLOB_PREFIX** | Yes |  | The name of the blob prefix which should be used for saved backup. Basically it should be the name of application for which backup is performed e.g. **service-catalog** |
```
Comment (dpolitesap) [inline tools/etcd-backup/README.md]: The name of the blob prefix to use to save the backup. Basically, it should be the name of the application for which the system performs the backup e.g. **service-catalog**

File: tools/etcd-backup/README.md [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+| **APP_BACKUP_ETCD_ENDPOINTS** | Yes |  | The endpoints of an etcd cluster. When multiple endpoints are given, the backup operator retrieves the backup from the endpoint that has the most up-to-date state. The given endpoints must belong to the same etcd cluster. |
```
Comment (dpolitesap) [inline tools/etcd-backup/README.md]: When there are multiple endpoints, the backup operator retrieves the backup from the endpoint that has the most up-to-date state.

File: tools/etcd-backup/main.go [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+		etcdOpInformersGroup  = etcdOpInformerFactory.Etcd().V1beta2()
+		etcdOpNsScopedLister  = etcdOpInformersGroup.EtcdBackups().Lister().EtcdBackups(cfg.WorkingNamespace)
+		etcdOpNsScopedCli     = etcdOperatorCli.EtcdV1beta2().EtcdBackups(cfg.WorkingNamespace)
+	)
+
+	// start informers
+	etcdOpInformerFactory.Start(stopCh)
+
+	// wait for cache sync
+	etcdOpInformerFactory.WaitForCacheSync(stopCh)
```
Comment (piotrmiskiewicz) [inline tools/etcd-backup/main.go]: question: do we need to setup a timeout for it?

**Comment by mszostok** [comment]
@dpolitesap thanks for the review, fixes applied :)

**Pull request merged**